camviewer_ui_impl.py

```diff
@@ -2456,31 +2503,76 @@ def onRemAvgEnter(self):
                 print("channel access exception: %s" % (e))
 
     def onReconnect(self):
-        self.timeoutdialog.reconn()
+        self.onCameraSelect(self.ui.comboBoxCamera.currentIndex())
 
     def onForceDisco(self):
         if self.cameraBase != "" and not self.haveforce:
             self.forcedialog = forcedialog(self.activedir + self.cameraBase + "/", self)
             self.haveforce = True
 
+    def update_display_timer(self, msec: int | None = None):
+        """
+        Show the user how much time until the rate timer expires.
+
+        If the timer is not active, this will set the display text to "Never",
+        or to "Timed Out" if we've completely timed out.
+
+        Otherwise, it will set the display text in the most appropriate form
+        from the options:
+
+        - x days, y hours
+        - x hours, y mins
+        - x mins
+        - under 1 minute
+        """
+        if msec is None:
+            if not self.discoTimer.isActive():
+                if self.last_des_max_rate <= 1:
+                    text = "Never"
+                else:
+                    text = "Timed Out"
+                self.ui.label_rate_timeout_value.setText(text)
+                return
+            msec = self.discoTimer.remainingTime()
+        sec = msec // 1000
+        mins, sec = divmod(sec, 60)
+        hours, mins = divmod(mins, 60)
+        days, hours = divmod(hours, 24)
+
+        if days:
+            text = f"{days} days, {hours} hours"
+        elif hours:
+            text = f"{hours} hours, {mins} mins"
+        elif mins:
+            text = f"{mins} mins"
+        else:
+            text = "under 1 minute"
+
+        self.ui.label_rate_timeout_value.setText(text)
+
     # We have been idle for a while!
     def do_disco(self):
-        self.discoTimer.stop()
-        self.timeoutdialog.activate()
+        self.stop_disco()
+        self.set_max_image_rate(1)
 
     def stop_disco(self):
         self.discoTimer.stop()
+        self.update_display_timer()
+
+    def refresh_disco(self):
+        if self.last_des_max_rate > 1:
+            self.discoTimer.start()
+            self.set_max_image_rate(self.last_des_max_rate)
 
     def setDisco(self, secs):
```

Comment:
A bit of disambiguation for myself (which I'd argue should be doc-stringed)
- do_disco: sets the app to its "idle state"
  - sets the max image rate to 1, stops the discoTimer and updates display (calls stop_disco)
- stop_disco: stops disco timer, updates display 
- refresh_disco: re-starts discoTimer

When DiscoTimer elapses we run `do_disco`, which sets the app into an idle (less frequently refreshing) state

Reply:
This seems to do what you detail on the PR description, I just wanted to make sure I understood the guts.

Reply:
I can install some docstrings for the next unfortunate maintainer